test/unit/repositories/test_catalogue_category.py

```diff
@@ -424,11 +432,19 @@ def test_delete_with_children_catalogue_items(test_helpers, database_mock, catal
     """
     catalogue_category_id = str(ObjectId())
 
-    # Mock count_documents to return 0 (children catalogue categories not found)
-    test_helpers.mock_count_documents(database_mock.catalogue_categories, 0)
-    # Mock count_documents to return 1 (children catalogue items found)
-    test_helpers.mock_count_documents(database_mock.catalogue_items, 1)
-
+    # Mock `find_one` to return no child catalogue category document
```

Comment:
All comments for mocking `find_one` should be updated so that they are like this rather than `... to return 1 (children catalogue categories found)` etc.